Support Onkyo Receivers via RS232 protocol.

rayzorben opened the pull request:
**Description:**
Add support for Onkyo Receivers via the RS232 serial port adapter.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2109

**Example entry for `configuration.yaml` (if applicable):**
```yaml
media_player:
     platform: onkyo_serial
     name: 'Onkyo TX-SR705'
     port: '/dev/ttyUSB0'
     zones:
         master:
             commands:
                 power:  'PWR'
                 volume: 'MVL'
                 source: 'SLI'
                 mute:   'AMT'
             queries:
                 power:  'PWRQSTN'
                 volume: 'MVLQSTN'
                 source: 'SLIQSTN'
                 mute:   'AMTQSTN'
         zone2:
             commands:
                 power:  'ZPW'
                 volume: 'ZVL'
                 source: 'SLZ'
                 mute:   'ZMT'
             queries:
                 power:  'ZPWQSTN'
                 volume: 'SVLQSTN'
                 source: 'SLZQSTN'
                 mute:   'ZMTQSTN'
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
    ** ran independent tests as other modules reported failure**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
      **unable to run due to error in openalpr, but no new requirements are generated**
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

molobrakos committed:
Handle UPS:es that returns more than one status (e.g. 'OL CHRG').

rayzorben committed:
Add Onkyo Serial support

rayzorben committed:
Add Onkyo Serial support

rayzorben committed:
Change URL

rayzorben committed:
Change URL

rayzorben committed:
added coverage

rayzorben committed:
added coverage

kellerza committed:
Allow 4.5min startup time for recorder

kellerza committed:
Allow 4.5min startup time for recorder

molobrakos committed:
Support for naming vehicle. Support for selecting what resources to display

rayzorben committed:
fixing lint/flake/pydoc issues

rayzorben committed:
fixing lint/flake/pydoc issues

homeassistant commented:
Hi @rayzorben, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6171).

Once you do that we will be able to review and accept this pull request.

Thanks!

pvizeli commented on homeassistant/components/media_player/onkyo_serial.py:
```diff
@@ -0,0 +1,323 @@
+from homeassistant.const import (STATE_OFF, STATE_ON, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = []
```
Please export all protocoll & device stuff to a external library. It is not allow to do that inside hass.

rayzorben replied:
This isn't any different than denon.py, itunes.py. From what I can tell, other components that do not do this have external libraries available on pypi or github which I do not.

Is there another option since moving this and hosting it in a full fledged external library is beyond the scope of a simple hass controller.

pvizeli replied:
You can make a mini library only for hass. `denon.py` have the logic on other side of telnet and use 10~ lines. REST are a grey area. Only simple calls are allow -> if I use `lib.do_stuff` or `request.get('/do/stuff')` is in the end the same and the logic is outside hass and the protocol is in request library. But I think also the iTunes are to big for include hass, but that is allready merge but I give a hint to Autor.

rayzorben replied:
No worries, I will split it out and update.

fabaff committed:
Upgrade fuzzywuzzy to 0.15.0 (#6175)

Add an optional extended description…

fabaff committed:
Upgrade slacker to 0.9.42 (#6173)

postlund committed:
Make it possible to ignore platforms in discovery (#6048)

* Make it possible to ignore platforms in discovery

* Add tests for discovery component

* small cleanups

molobrakos committed:
Merge pull request #6162 from molobrakos/nut

sensor.nut: Handle multiple statuses returned

molobrakos committed:
cache name in entity so we can display it even if the connection is lost (#6176)

update tellduslive dependency
check return value from update

molobrakos committed:
Merge pull request #6177 from molobrakos/volvooncall

volvooncall: support for naming vehicle and selecting what attributes to display

molobrakos committed:
volvooncall: icon and unit were mixed

molobrakos committed:
Merge pull request #6180 from home-assistant/molobrakos-patch-1

volvooncall: icon and unit were mixed

molobrakos committed:
's' is SI symbol for seconds (#6181)

colinodell committed:
QNAP update (#6182)

* Bump qnapstats library to 0.2.3

* Expose the new timeout setting

* Show persistent notification if QNAP fails to set up

This sitaution will usually occur if the server configuration is wrong or the
timeout is set too low.  In both cases `api.update()` will fail, which is a
problem because we cannot initialize many of the sensors without having this
data.

* Add new system_temp condition to QNAP sensor

pvizeli committed:
Homematic icon map / upper case on services (#6178)

* Set upper on service / add sensor icon mapper

* Add more icons

* fix id

pvizeli committed:
Add device_class to image_processing  (#6186)

* Add image_processing device_class

* Fix comments

* address comments

pvizeli committed:
Cleanup run_callback_threadsafe (#6187)

* Cleanup run_callback_threadsafe

* fix spell

* Revert image_processing, they need to wait for update

colinodell committed:
Use H2 headers to split up the different sections (#6183)

Using headers makes it easier to visually differentiate between the different sections

pvizeli committed:
Refactory of envisalink (#6160)

* Refactory of envisalink

* remove event buss

* init dispatcher from hass.

* Move platform to new dispatcher

* fix lint

* add unittest & threadded functions

* fix copy & past error

pvizeli committed:
Refactory of envisalink (#6160)

* Refactory of envisalink

* remove event buss

* init dispatcher from hass.

* Move platform to new dispatcher

* fix lint

* add unittest & threadded functions

* fix copy & past error

andrey-git committed:
Create zwave devices on OZW thread and only add them during discovery (#6096)

* Create zwave devices on OZW thread and only add them during discovery.

* Read and write devices dict from loop thread.

* More async

* replace callback with coroutine

* import common function instead of callin git

kellerza committed:
Bugfix restore startup state (#6189)

kellerza committed:
Bugfix restore startup state (#6189)

balloob committed:
Random test fixes (#6195)

* Store persistent errors in hass (speeds up tests)

* Fix sleepiq test dependency on test order

* Fix sleepiq validation

balloob committed:
Extend test for group config

balloob committed:
Remove automatically reloading group config (#6197)

balloob committed:
Remove automatically reloading group config (#6197)

balloob committed:
Default config to setup group editor (#6198)

Danielhiversen committed:
minor broadlink fix (#6202)

lindsaymarkward committed:
Update Yeelight Sunflower light platform to 0.0.6 (#6208)

Add an optional extended description…

lindsaymarkward committed:
Update Yeelight Sunflower light platform to 0.0.6 (#6208)

Add an optional extended description…

andrey-git committed:
Some zwave cleanup (#6203)

molobrakos committed:
sensor.speedtest: provide a default icon (#6207)

aronsky committed:
Test the temperature returned by RM2 (#6205)

* Test the temperature returned by RM2
* Validate fields via voluptuous
* Fixed range for humidity

Zac-HD committed:
Zamg weather (#5894)

* Fast & efficient updates for ZAMG weather data

ZAMG updates on the hour, so instead of checking every half-hour we can
check each minute - only after the observations are taken until
receiving them.

* sensor.zamg: test instead of whitelist for station_id

* Autodetect closest ZAMG station if not given

* ZAMG weather component, based on the sensor

* Review improvements

* Update to new ZAMG schema, add logging

Turns out it wasn't a typo, but rather an upstream schema change.  Added
better error handling to ease diagnosis in case it happens again.

* No hardcoded name

pavoni committed:
Bump pyloopenergy - catch socketIO exceptions.

pavoni committed:
Merge pull request #6213 from home-assistant/bump_pyloopenergy

Bump pyloopenergy - catch socketIO exceptions.

balloob committed:
Fix reporting on bad login (#6201)

balloob committed:
Fix reporting on bad login (#6201)

pvizeli committed:
Move mqtt from eventbus to dispatcher / add unsub for dispatcher  (#6206)

* Move mqtt from eventbus to dispatcher / add unsub for dispatcher

* Fix lint

* Fix test

* Fix lint v2

* fix dispatcher_send

andrey-git committed:
Update flake8 and pylint to latest (#6217)

molobrakos committed:
sensor.dovado: Upgraded library version

molobrakos committed:
sensor.eliqonline: Change icon

fabaff committed:
Fix link (#6219)

kellerza committed:
No wait for start and more async

kellerza committed:
No wait for start and more async

balloob committed:
Version bump to 0.39

andrey-git committed:
Make glob preserve order (#6224)

andrey-git committed:
Make glob preserve order (#6224)

fabaff committed:
Update regex (#6216)

fabaff committed:
Update regex (#6216)

balloob committed:
Merge pull request #6199 from home-assistant/release-0-39

0.39

balloob committed:
Merge pull request #6167 from kellerza/recorder_timer

Recorder async & remove start_recording event

balloob committed:
Update frontend

balloob committed:
Update frontend

balloob committed:
Merge pull request #6227 from home-assistant/release-0-39

Release 0 39

balloob committed:
Fix recorder async (#6228)

balloob committed:
Fix recorder async (#6228)

balloob committed:
Merge pull request #6229 from home-assistant/release-0-39

Fix recorder async (#6228)

pschmitt committed:
Fix livebox-play interactions for Python < 3.6 (#6243)

balloob committed:
Ensure we properly close HASS instances. (#6234)

postlund committed:
Add service to change log levels (#6221)

* Add service to change log levels

* Fix review comments

pvizeli committed:
Move ffmpeg to dispatcher from hass.data entity store. (#6211)

* Move ffmpeg to dispatcher from hass.data entity store.

* fix lint

* address paulus comments

* add more unittest for better coverage

balloob committed:
Feature/reorg recorder (#6237)

* Re-organize recorder

* Fix history

* Fix history stats

* Fix restore state

* Lint

* Fix session reconfigure

* Move imports around

* Do not start recording till HASS started

* Lint

* Fix logbook

* Fix race condition recorder init

* Better reporting on errors

pvizeli committed:
Bugfix mqtt socket error (#6256)

pvizeli committed:
Bugfix mqtt socket error (#6256)

shenning00 committed:
Notify ciscospark (#6130)

* Adding ciscospark notifier

* Adding ciscospark notifier

* CI cleanup.

* houndci-bot changes

* ok --- a bunch of code verify changes

balloob committed:
Config fix (#6261)

balloob committed:
Config fix (#6261)

pvizeli committed:
Bugfix mqtt paho client to speend time (#6266)

pvizeli committed:
Bugfix mqtt paho client to speend time (#6266)

balloob committed:
Version bump to 0.39.1

balloob committed:
Merge pull request #6272 from home-assistant/release-0-39-1

0.39.1

jawilson committed:
Properly report features for each hue bulb type (#6271)

jjmontesl committed:
Local file camera now supports yet inexisting files. (#6157)

BillyNate committed:
light.transition now supports float instead of int in order to be able to perform faster transitions (#6163)

groth-its committed:
Fix for OSRAM lights connected to hue bridge (#6122)

* Fix for OSRAM lights connected to hue bridge

Do not send command "effect = none" to OSRAM lights
Osram lights connected to a hue bridge do not seem to handle "effect =
none" very well. Most of the times they jump to the selected color and
then change to red within a second.

Osram lights connected to a hue bridge do not handle xy values outside
of their gamut. Since they just stay at their old color value, handling
the UI is very unpredictable. Sending HSV values to the lights fixes this.

* Add tests for new util methods

BastianPoe committed:
Add support for MAX!Cube thermostats and window shutter sensors (#6105)

vroomfonde1 committed:
Analog modem callerid support (#5840)

* analog-modem-callerid

* analog-modem-callerid

* analog-mod

* Updates from latest review

* Updates from latest review

rayzorben committed:
updated to move library external

rayzorben committed:
updated to move library external

rayzorben commented:
Please re-review as library has been moved externally. I have never done an updated pull request before, and the changes show in the commits as part of this pull request, but let me know if this is not correct. I have also updated the documentation.

rayzorben commented on homeassistant/components/media_player/onkyo_serial.py:
```diff
@@ -0,0 +1,198 @@
+        if sources:
+            self._sources = dict(self._sources.items() + sources.items())
+        self._source_list = list(self._sources.values())
+        #pylint: disable=E
```
this looks like a bug on houndci-bot, but the disable is because pylint complains that Instance of 'OnkyoSerial' has no 'on_state_change' member when it does and is working fine.

pvizeli replied:
Need to be `# pylint`

pavoni committed:
Fix vera thermostat mode set bug

danielperna84 committed:
[sensor.dnsip] New Sensor: DNS IP (#6214)

* Added DNS IP sensor

* Removed unused import

* Added coverage

* fixed flake

* Applied suggested changes

* Removed debug code

* Switched to aiodns

* Raised scan interval

* Updating state with entity creation

* Lint

* Updated requirements_all

pavoni committed:
Merge pull request #6278 from home-assistant/fix_vera_thormostat_bug

Fix vera thermostat bug

lindsaymarkward committed:
Update library version for Yeelight Sunflower lights platform (fix for packaging problem with 0.0.7) (#6233)

arjenfvellinga committed:
Prevent duplicate names on Vera devices by appending the device id (#6100)

* Prevent duplicate names by prepending device id to it.

* Always append device id, not conditionally.

* Moved naming of devices

* flake8

andrey-git committed:
Add temperature support for MH-Z19 CO2 sensor. (#6169)

* Add temperature support for MH-Z19 CO2 sensor.

* Remove debug printout

* More tests

* Minor fixes

bokub committed:
improve history_stats accuracy (#6294)

alanfischer committed:
Updated pyitachip2ir (#6296)

* Updated pyitachip2ir

* Updated requirements_all.txt

Commit:
Influx fix (#6289)

* Fix: replace influxdb query by another query that is more lightweight and won't timeout

* Fix: replace influxdb query by another query that is more lightweight and won't timeout

emlove committed:
Fix toggle and media_play_pause post async (#6291)

pvizeli committed:
Migrate calendar setup to async. (#6305)

zhelev committed:
Frontier silicon (#6131)

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* changed info to debug

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added the frontier_silicon component

* trying to satisfy pylint

* fsapi version 0.0.6

* remove white space

* generated requirements

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* trying to satisfy pylint

* changed info to debug

* added the frontier_silicon component

* fsapi version 0.0.6

* generated requirements

* pylint

* moved import requests to the method where it is being used

* add a basic unit test

* cleaned up source code

* added frontier_silicon constant

* added the frontier_silicon component

* added basic test

* added fsapi to requirements_all.txt

* added coverage omit, though a basic test was included

* added MEDIA_TYPE_MUSIC for artist and album

* removed duplicate cons

* switched fsapi call to a property, removed unecessary comment

* detailed docstring for fs_device

* added a space for the info_name - info_text separator

* reduced proeprty (fsapi) access for volume down/up

emlove committed:
Version bump to 0.40.0.dev0

pvizeli committed:
Bootstrap / Component setup async (#6264)

* Bootstrap / Entiy setup async

* Cleanup add_job stuff / return task/future object

* Address paulus comments / part 1

* fix install pip

* Cleanup bootstrap / move config stuff to config.py

* Make demo async

* Further bootstrap improvement

* Address Martin's comments

* Fix initial tests

* Fix final tests

* Fix bug with prepare loader

* Remove no longer needed things

* Log error when invalid config

* More cleanup

* Cleanups platform events & fix lint

* Use a non blocking add_entities callback for platform

* Fix Autoamtion is setup befor entity is ready

* Better automation fix

* Address paulus comments

* Typo

* fix lint

* rename functions

* fix tests

* fix test

* change exceptions

* fix spell

e850205 committed:
Convert kpH and mpH to kph and mph (#6316)

There is no reason for the H to be capitalized. Changing it to lowercase increases consistency with other components and allows for use of the min/max sensor, which throws an error if the units of measurement are not the same.

deftdawg committed:
Rollback netdisco to 0.8.2 to resolve #6165 (#6314)

* Rollback netdisco to 0.8.2 to resolve #6165

* Rollback netdisco to 0.8.2 to resolve #6165

kellerza committed:
Log errors when loading yaml (#6257)

balloob committed:
Bootstrap tweaks tests (#6326)

* Update strings/fix component not found message.

* Fix tests

* More tweak text

molobrakos committed:
Merge pull request #6328 from molobrakos/eliqonline

sensor.eliqonline: Change to more appropriate icon

molobrakos committed:
Merge pull request #6329 from molobrakos/dovado

sensor.dovado: Upgraded library version

scipioni committed:
Telegram webhooks new text event (#6301)

* new TELEGRAM_TEXT

* telegram command event renamed in 'telegram_command'

* fire telegram_text event anyway

pvizeli committed:
Cleanup component track_point_in_utc_time usage (#6330)

balloob committed:
Discovery fix (#6321)

* Fix incorrect import

* Create own discovery service

* Fix tests

* Fix hdmi_cec bad import

balloob committed:
Discovery fix (#6321)

* Fix incorrect import

* Create own discovery service

* Fix tests

* Fix hdmi_cec bad import

balloob committed:
Test against 3.6-dev (#6324)

pvizeli committed:
Bugfix ZigBee / Move from eventbus to dispatcher (#6333)

* Bugfix ZigBee / Move from eventbus to dispatcher

* fix lint

pvizeli committed:
Bugfix ZigBee / Move from eventbus to dispatcher (#6333)

* Bugfix ZigBee / Move from eventbus to dispatcher

* fix lint

balloob committed:
Bump netdisco to 0.9.1 (#6338)

balloob committed:
Bump netdisco to 0.9.1 (#6338)

balloob committed:
Version bump to 0.39.2

rayzorben reviewed (commented):
I have made the requested changes, thanks.

balloob committed:
Merge pull request #6339 from home-assistant/release-0-39-2

0.39.2

molobrakos committed:
sensor.dovado: compute state in update (#6340)

MartinHjelmare committed:
Fix mysensors callback race (#6311)

* Fix possible race at startup in mysensors callback

* Update devices via persistence before starting gateway to avoid
  two threads calling the same callback at the same time.

* Call add_devices max once per callback

fabaff committed:
Upgrade TwitterAPI to 2.4.5 (#6351)

fabaff committed:
Upgrade py-cpuinfo to 0.2.6 (#6335)

pvizeli committed:
Template sensor change flow / add restore (#6336)

andrey-git committed:
Zwave optimize value_added (#6210)

* Make zwave devices listen on less network changes.

* Convert more platforms

* Remove printouts.

* Fix copy-paste

* Change default dependent list to empty list

Commit:
Update Vagrant provision.sh (#6236)

- Bugfix: with f63a79ee we removed `script/home-assistant@.service`
  systemd unit file, which is used by Vagrant box to start/stop hass
- simplify interaction with Vagrant, provision.sh now is the only
  entry point and doesn't need the user to touch/remove files in
  order to change provisioner behavior

masarliev committed:
Update Adafruit_Python_DHT to support new raspberry kernel (#6325)

* Update Adafruit_Python_DHT to support new raspberry kernel

* update Adafruit Python DHT

reedriley committed:
Add fallback for name if userdevicename isn't set using old serialnumber logic (#6265)

Add an optional extended description…

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

titilambert committed:
Improve Honeywell US climate component (#5313)

* Improve Honeywell US climate component

* Fix tests

* Fix tests

* Add cool_away_temp and heat_away_temp for honeywell US

* Fix honeywell tests

* Fix PR comments

pvizeli committed:
Template binary_sensor change flow / add restore (#6343)

* Template binary_sensor change flow / add restore

* fix lint

Duoxilian committed:
Additional support for ecobee hold mode (#6258)

* Integrate suggestion in #5590 by nordlead2005. This change has been
sitting in limbo for over a month, but it is a good idea. I don't
mean to step on nordlead2005's toes, but we need to make progress.

* Use defined constant for TEMPERATURE_HOLD

* Integrate handling of vacation into hold mode. Canceling vacation
hold requires an update to the external pyecobee library. Creation
of vacation is not supported (it would be straightforward in the code,
but a complex user interface would be required, similar to what is
now done in the ecobee thermostat).

* Add capability to retrieve list of defined climates from ecobee.

* The mode() method used to return the system mode in internal
representation. However, the user sees a different notation in
the ecobee thermostat. Seeing some internal name is particularly
weired with user-defined climates, where these are named "smart1",
"smart2", etc., instead of the name the user has defined. Return
the user-defined name instead. This change might break some user
interfaces but is easily remedied (e.g., use "Away" instead of
"away").

* Simplify is_away_mode_on().

* Correction of erroneously indented else statement.

* Change comment as flake8 gets confused.

dramamoose committed:
Update Formulas in Convert XY to RGB (#6322)

* Update to Current RGB D65 Conversion

As per Philips Hue https://developers.meethue.com/documentation/color-conversions-rgb-xy

* Update the source of the XYZ to RGB formulas

* Fix Whitespace

* Update Whitespace

* Update Tests for new Formulas

* Update Tests

* Update XY_Brightness_to_hsv tests

* Update test_color.py

balloob committed:
Update coveragerc

emlove committed:
Use dynamic ports for test instances (#6232)

martinfrancois committed:
Added support for multiple codes executed in a row (#5908)

* Added support for multiple codes executed in a row

now codes can be specified either by simply providing a single code, which will then be sent like usual, or multiple codes can be executed in a row, specified in a comma delimited format in the configuration.yaml. For example: 111111,222222,333333,444444 would mean 111111 would be sent first, followed by 222222 and 333333 and 444444.

* rpi_rf: added line breaks to not exceed 79 characters per line

* include validation for correct formatting of codes

added regex which only allows either a single number (like 1252456245) or a sequence of commas followed by another number.

* added line breaks to not exceed 79 characters per line

* fix for 'continuation line under-indented for visual indent'

* another try at 'continuation line under-indented for visual indent'

* changed from regex to list for easier maintainability

* removed unnecessary splitting of strings

postlund committed:
Use push updates in Apple TV (#6323)

* Use push updates in Apple TV

* Fix review comments

jawilson committed:
Fix command sudo not found error in dev Dockerfile (#6346)

pvizeli commented on homeassistant/components/media_player/onkyo_serial.py:
```diff
@@ -0,0 +1,198 @@
+        #pylint: disable=E
+        self._device = OnkyoSerial(config, zone, port=port)
+        self._device.on_state_change += self.state_changed
+        self.update()
```
Please remove it and use `add_devices([], True)`

rayzorben replied:
i assume you mean just the update. and passing True calls the update, so changed.

pvizeli commented on homeassistant/components/media_player/onkyo_serial.py:
```diff
@@ -0,0 +1,198 @@
+
+DEFAULT_NAME = 'Onkyo Serial Receiver'
+DEFAULT_PORT = '/dev/ttyUSB0'
+DEFAULT_ZONES = load("""
```
Please do create a real dict and not use yaml for that simple thing.

rayzorben replied:
changed, since i am providing default yaml for the configuration, my method seems more consistent and readable, but updated anyway.

pvizeli reviewed (changes requested):
Some lines have CRLN instead LN. Please fix that.

alanfischer committed:
Fix calendar authentication text, and handle calendar events without summaries. (#6337)

* Fixed google authorization text

* Let calendar handle events without a summary

pvizeli committed:
Move dispatcher out of init. (#6355)

andrey-git committed:
Zwave: Add remove/replace failed node services. (#6248)

* Zwave: Add remove/replace failed node services.

* Fix text

pvizeli committed:
Template switch change flow / add restore (#6356)

* Template switch change flow / add restore

* fix tests

* fix binary_sensor template

janLo committed:
Bump limitlessled dependency to 1.0.5. (#6334)

This fixes issue #6295.

milaq committed:
snmp: upgrade pysnmp to 4.3.4 (#6359)

* snmp: upgrade pysnmp to 4.3.4

fixes https://github.com/home-assistant/home-assistant/issues/6238

* snmp: v4.3.4: add missing definition changes

pvizeli committed:
Bugfix new async_add_devices function (#6362)

pvizeli committed:
Restore for input_slider (#6360)

rayzorben committed:
code review changes

rayzorben committed:
code review changes

rayzorben committed:
fix pylint complaint

rayzorben committed:
fix pylint complaint

rayzorben reviewed (commented):
done

Commit:
Added IPv4 data collector (#6304)

* Added IPv4 data collector

* Formatting

* Bugfix: data is in kBit/s not kByte/s

jjmontesl committed:
Return None instead of raising ValueException from as_timestamp template function. (#6155)

kellerza committed:
[recorder] Catch more startup errors #6179 (#6192)

* [recorder] Catch more startup errors #6179

* Rebase on new recorder

happyleavesaoc committed:
twilio component (#6348)

* twilio component

* add http dependency to twilio

* fire->async_fire

andrey-git committed:
Add Z-Wave battery level as a sensor. (#6341)

pvizeli committed:
OwnTrack Async (#6363)

* Migrate owntrack to async

* fix tests

pvizeli committed:
Fix possibility that have multible topic subscribe mqtt (#6372)

pvizeli committed:
Migrate mqtt tracker and arwn sensor to async / cleanup owntrack (#6373)

* Migrate mqtt tracker and arwn sensor to async / cleanup owntrack

* Fix tests / lint

balloob committed:
Z-Wave prevent I/O event loop (#6369)

* Prevent Z-Wave I/O in event loop

* Move value_handler to util class.

* Add docstring

valentinalexeev committed:
Update pwaqi to 3.0 to use public API (#6376)

The underlying PWAQI library version 3.0 is now using public API to
access AQICN data.

mezz64 committed:
Update Hikvision Binary Sensors to latest library, remove pyDispatcher (#6231)

* Update pyHik version, remove pyDispatcher in favor of callbacks

* Fix naming

* Fix lint blank line

* Move stream thread start to HOMEASSISTANT_START event

* Bump library version to cleanup shutdown

* Fix requirements

pvizeli commented on homeassistant/components/media_player/onkyo_serial.py:
```diff
@@ -0,0 +1,203 @@
+SOURCES_SCHEMA = vol.Schema({vol.All(cv.string): vol.Any(cv.string)})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT): DEFAULT_PORT,
```
please change that to `vol.Required(CONF_PORT): cv.isdevice`

pvizeli reviewed (commented):
Please address my last comment and I'll merge it. You need also rebase your branch (https://home-assistant.io/developers/development_catching_up/) at the moment you stay in conflict with dev.

rayzorben committed:
updating schema for device

rayzorben committed:
updating schema for device

rayzorben committed:
rebase

rayzorben reviewed (commented):
i changed to isdevice as requested, and tried to rebase. i dont know if it is expected now to show 280 files changed or if i have messed something up. if everything looks good please proceed or let me know what i messed up please.

rayzorben reviewed (commented):
fixed.

rayzorben commented:
build is failing due to missing requirements in requirements_all.txt. I am unable to run gen_requirements

```bash
[19:31:33] pi@home:~/development/home-assistant$ script/gen_requirements_all.py
Traceback (most recent call last):
  File "script/gen_requirements_all.py", line 156, in <module>
    main()
  File "script/gen_requirements_all.py", line 139, in main
    data = gather_modules()
  File "script/gen_requirements_all.py", line 75, in gather_modules
    for package in sorted(explore_module('homeassistant.components', True) +
  File "script/gen_requirements_all.py", line 39, in explore_module
    module = importlib.import_module(package)
  File "/usr/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2212, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2224, in _find_and_load_unlocked
ImportError: No module named 'homeassistant'
```

rayzorben commented:
@pvizeli any idea why I can't run gen_requirements_all.py - i am getting the error above. Also is the rebase I did supposed to show all of the changes listed above?

rayzorben commented:
Apologies, maybe I messed up, but trying to get this back to square one, fix gen_rqeuirements_all and resubmit.
